site/content/extend/plugins/signing-plugins.md

```diff
@@ -0,0 +1,151 @@
+---
+title: "Signing Plugins"
+date: 2019-09-04T00:00:00-05:00
+subsection: Plugins (Beta)
+weight: 50
+---
+
+"Plugin Marketplace” is a platform used to distribute common plugins where Mattermost admins can install plugins without verification. To trust a plugin, there needs to be a way to verify the authenticity and integrity of a plugin before being installed. [The GNU Privacy Guard (GPG)](https://gnupg.org) tool is used to sign and verify plugins. Every plugin is signed using a private key and every Mattermost instance should have access to a public key which is used to verify plugin signatures on installation.
+
+## Key Generation
+Public and private key pairs are needed to sign and verify plugins. Private key is used for signing and should be kept in the Vault. Public key is used for verification and can be distributed freely. To generate a key pair, run the following command:
```

Comment:
Let's avoid assuming a vault, and just talk about keeping the private key secure.